packages/authorization-process/src/routers/AuthorizationRouter.ts

```diff
@@ -118,6 +118,56 @@ const authorizationRouter = (
         }
       }
     )
+    .get(
+      "/clientsWithKeys",
+      authorizationMiddleware([
+        ADMIN_ROLE,
+        SECURITY_ROLE,
+        M2M_ROLE,
+        SUPPORT_ROLE,
+      ]),
+      async (req, res) => {
+        const ctx = fromAppContext(req.ctx);
+        try {
+          const { name, userIds, consumerId, purposeId, kind, offset, limit } =
+            req.query;
+          const clients = await authorizationService.getClients({
+            filters: {
+              name,
+              userIds: userIds?.map(unsafeBrandId<UserId>),
```

Comment:
if the requester role is `SECURITY_ROLE` we should filter only for the requester id

Reply:
Updated